hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -335,7 +335,21 @@ int MDMParser::sendFormattedWithArgs(const char* format, va_list args) {
 bool MDMParser::_atOk(void)
 {
     sendFormated("AT\r\n");
-    return (RESP_OK == waitFinalResp(nullptr, nullptr, AT_TIMEOUT));
+    bool atok = (RESP_OK == waitFinalResp(nullptr, nullptr, AT_TIMEOUT));
+
+    // R410 Power Savings Mode currently disabled, therefore this only affects all other modems
+    if (!atok && _dev.dev != DEV_SARA_R410 && _dev.lpm == LPM_ACTIVE) {
```

Comment:
Instead of testing against a specific modem, perhaps we could test against a capability.

> Gen 2 & Gen 3 LTE devices are not affected due to power saving mode being disabled by default.

What if power saving mode has been enabled on these platforms, does this logic account for that situation?

Reply:
No it does not, but we shouldn't need to add that logic until we enable power saving mode on those devices (currently it's hard coded to be disabled)

Reply:
Ahh. Okay, I didn't realize it was hardcoded.

However, I do think if we don't account for it now, then we will forget it once we enable power saving mode, and have a hard to find bug at that time.

Reply:
I added TODO comments near the code that hard codes the disabling of the power modes.  Since the behavior of Power Saving mode for R410 is different than 2G/3G, it is not possible to code this logic in now without fully developing this feature for R410 (which we won't do at this time).